core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala

```diff
@@ -36,7 +36,7 @@ import org.apache.spark.internal.config.Python.PYSPARK_EXECUTOR_MEMORY
  * This is meant to be immutable so user can't change it after building.
  */
 @Evolving
-class ResourceProfile(
+private[spark] class ResourceProfile(
```

Comment:
Let's remove `@Evolving` for `private[spark]`.

Reply:
same as above